lib/grammers-client/src/client/files.rs

```diff
@@ -9,16 +9,27 @@
 use crate::types::{Media, Uploaded};
 use crate::utils::generate_random_id;
 use crate::Client;
+use futures_util::future::try_join_all;
 use grammers_mtsender::InvocationError;
 use grammers_tl_types as tl;
-use std::io::SeekFrom;
-use std::path::Path;
-use tokio::fs;
-use tokio::io::{self, AsyncRead, AsyncReadExt as _, AsyncSeekExt as _, AsyncWriteExt as _};
+use std::{
+    io::SeekFrom,
+    path::Path,
+    sync::{
+        atomic::{AtomicI32, Ordering},
+        Arc,
+    },
+};
+use tokio::{
+    fs,
+    io::{self, AsyncRead, AsyncReadExt as _, AsyncSeekExt as _, AsyncWriteExt as _},
+    sync::Mutex,
+};
 
 pub const MIN_CHUNK_SIZE: i32 = 4 * 1024;
 pub const MAX_CHUNK_SIZE: i32 = 512 * 1024;
 const BIG_FILE_SIZE: usize = 10 * 1024 * 1024;
+const WORKER_COUNT: usize = 4;
```

Comment:
This should be configurable. With the work done by #72, it could maybe be configurable (although we can change that later).